app/src/main/java/com/rajatgoyal/puzzle15/ui/GameActivity.java

```diff
@@ -513,62 +523,70 @@ public void onClick(DialogInterface dialog, int which) {
     }
 
 
-    private void swipeHandler(View view, SWIPE DIR) {
-        int i, j = 0;
-
-        // Get button's coordinates using id matrix
-        label:
-        for (i = 0; i < size; i++) {
-            for (j = 0; j < size; j++) {
-                if (view.getId() == id[i][j])
-                    break label;
-            }
-        }
+    private void swipeHandler(SWIPE DIR) {
+        int i, j;
+        Timber.d("Swipe Handler called");
 
-        String buttonText = buttons[i][j].getText().toString();
-        if (TextUtils.isEmpty(buttonText)) {
-            // user clicked on the empty tile
-            return;
-        }
-
-        int num = Integer.parseInt(buttonText);
-        int i1 = i, j1 = j;
-
-        if (DIR.equals(SWIPE.RIGHT) && isSwipeValid(i, j + 1)) {
+        /*
+          Check if the 4 button around the empty cell is valid or not corresponding to the
+          swipe direction
+         */
+        if (DIR.equals(SWIPE.RIGHT) && isSwipeValid(emptyRowIndex, emptyColIndex - 1)) {
```

Comment:
consider moving the code which is different for different swipe to their respective functions and leave common code here

Reply:
Done in 97916924f16e00d0530397dfcaf672ec7cd12bd5